### PR TITLE
ER-1306 Moving Training Provider to Part 1

### PR DESCRIPTION
**Pull request by chrisjensenuk**
Moving Training Provider from Part 2 to Part 1

There is now an extra question on the page that allows a user to skip the specifying of the Training Provider.

### CODE REVIEW

File: src/Employer/Employer.Web/ViewModels/Part1/TrainingProvider/SelectTrainingProviderViewModel.cs [hunk 1]
```diff
@@ -1,17 +1,16 @@
...
 {
     public class SelectTrainingProviderViewModel
     {
         public string Title { get; set; }
-        [Display(Name = "UKPRN")]
         public string Ukprn { get; set; }
         public string TrainingProviderSearch { get; set; }
-        public string SelectedTrainingText { get; set; }
         public IEnumerable<string> TrainingProviders { get; set; }
+        public bool? SelectTrainingProvider { get; set; }
```
Comment (dev4valtech) [inline src/Employer/Employer.Web/ViewModels/Part1/TrainingProvider/SelectTrainingProviderViewModel.cs]: IsTrainingProviderSelected

File: src/Employer/Employer.Web/ViewModels/Part1/TrainingProvider/SelectTrainingProviderEditModel.cs [hunk 2]
```diff
@@ -9,6 +9,8 @@ public class SelectTrainingProviderEditModel : VacancyRouteModel
         public string TrainingProviderSearch { get; set; }
 
         public TrainingProviderSelectionType SelectionType { get; set; }
+
+        public bool? SelectTrainingProvider { get; set; }
```
Comment (dev4valtech) [inline src/Employer/Employer.Web/ViewModels/Part1/TrainingProvider/SelectTrainingProviderEditModel.cs]: IsTrainingProviderSelected or HasSelectedTrainingProvider

File: src/Employer/Employer.Web/Views/Part1/TrainingProvider/SelectTrainingProvider.cshtml [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+                        <label class="govuk-label govuk-radios__label" for="found_yes_conditional">
+                            Yes
+                        </label>
+                    </div>
+                    <div class="govuk-radios__conditional" id="found_yes_conditional">
+                        <div class="govuk-!-margin-bottom-2">
+                            <noscript>
+                                <div class="govuk-form-group" id="non-js-search">
+                                    <span class="govuk-hint">
+                                        Enter the training provider reference number (UKPRN). You will be able to get this from your training provider The UKPRN must only be 8 digits (For example, 12345678)
```
Comment (dev4valtech) [inline src/Employer/Employer.Web/Views/Part1/TrainingProvider/SelectTrainingProvider.cshtml]: > your training provider The UKPRN must 
is there a punctuation missing here?

File: src/Employer/Employer.Web/Views/Part1/TrainingProvider/SelectTrainingProvider.cshtml [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+                                    </span>
+                                    <span esfa-validation-message-for="Ukprn" class="govuk-error-message"></span>
+                                    <input asp-for="Ukprn" type="text" maxlength="8" class="govuk-input govuk-input--width-10" />
+                                </div>
+                            </noscript>
+                            <div class="govuk-form-group js-hidden" id="js-search">
+                                <span class="govuk-hint">
+                                    Enter the training provider or reference number (UKPRN). <br>
+                                    UKPRN is 8 numbers. For example 12345678.
```
Comment (dev4valtech) [inline src/Employer/Employer.Web/Views/Part1/TrainingProvider/SelectTrainingProvider.cshtml]: on Line 40 we have used 8 digits and here 8 numbers.

File: src/Employer/Employer.Web/ViewModels/Part1/TrainingProvider/SelectTrainingProviderEditModelValidator.cs [hunk 5]
```diff
@@ -1,13 +1,18 @@
...
     public class SelectTrainingProviderEditModelValidator : AbstractValidator<SelectTrainingProviderEditModel>
     {
         public SelectTrainingProviderEditModelValidator()
         {
-            When(m => m.SelectionType == TrainingProviderSelectionType.Ukprn, () =>
+            RuleFor(m => m.SelectTrainingProvider)
+                .NotNull()
+                .WithMessage("Please select an option to continue");
+
+            When(m => m.SelectTrainingProvider == true &&
```
Comment (sachpatel) [inline src/Employer/Employer.Web/ViewModels/Part1/TrainingProvider/SelectTrainingProviderEditModelValidator.cs]: If you rename this property I don't think you will need the `== true`

File: src/Employer/Employer.Web/Orchestrators/Part1/TrainingProviderOrchestrator.cs [hunk 6]
```diff
@@ -17,8 +16,22 @@
 using Esfa.Recruit.Vacancies.Client.Infrastructure.Services.TrainingProvider;
 using Microsoft.Extensions.Logging;
 
-namespace Esfa.Recruit.Employer.Web.Orchestrators.Part2
+namespace Esfa.Recruit.Employer.Web.Orchestrators.Part1
 {
+    public enum PostSelectTrainingProviderResultAction
```
[the remark this answers is not in the document]
Comment (chrisjensenuk) [reply src/Employer/Employer.Web/Orchestrators/Part1/TrainingProviderOrchestrator.cs]: 👍

File: src/Employer/Employer.Web/ViewModels/Part1/TrainingProvider/SelectTrainingProviderViewModel.cs (hunk 1 above)
Comment (chrisjensenuk) [reply src/Employer/Employer.Web/ViewModels/Part1/TrainingProvider/SelectTrainingProviderViewModel.cs]: 👍

File: src/Employer/Employer.Web/ViewModels/Part1/TrainingProvider/SelectTrainingProviderEditModel.cs (hunk 2 above)
Comment (chrisjensenuk) [reply src/Employer/Employer.Web/ViewModels/Part1/TrainingProvider/SelectTrainingProviderEditModel.cs]: 👍

File: src/Employer/Employer.Web/Views/Part1/TrainingProvider/SelectTrainingProvider.cshtml (hunk 3 above)
Comment (chrisjensenuk) [reply src/Employer/Employer.Web/Views/Part1/TrainingProvider/SelectTrainingProvider.cshtml]: 👍

File: src/Employer/Employer.Web/Views/Part1/TrainingProvider/SelectTrainingProvider.cshtml (hunk 4 above)
Comment (chrisjensenuk) [reply src/Employer/Employer.Web/Views/Part1/TrainingProvider/SelectTrainingProvider.cshtml]: 👍  Content is a bit of a mess. I'll catch up with Helen.

File: src/Employer/Employer.Web/ViewModels/Part1/TrainingProvider/SelectTrainingProviderEditModelValidator.cs (hunk 5 above)
Comment (chrisjensenuk) [reply src/Employer/Employer.Web/ViewModels/Part1/TrainingProvider/SelectTrainingProviderEditModelValidator.cs]: Its a nullable bool so this needs to be explicit

File: src/Employer/Employer.Web/ViewModels/Part1/TrainingProvider/SelectTrainingProviderEditModelValidator.cs [hunk 7]
```diff
@@ -1,13 +1,18 @@
...
+namespace Esfa.Recruit.Employer.Web.ViewModels.Part1.TrainingProvider
 {
     public class SelectTrainingProviderEditModelValidator : AbstractValidator<SelectTrainingProviderEditModel>
     {
         public SelectTrainingProviderEditModelValidator()
         {
-            When(m => m.SelectionType == TrainingProviderSelectionType.Ukprn, () =>
+            RuleFor(m => m.SelectTrainingProvider)
+                .NotNull()
+                .WithMessage("Please select an option to continue");
```
[the remark this answers is not in the document]
Comment (chrisjensenuk) [reply src/Employer/Employer.Web/ViewModels/Part1/TrainingProvider/SelectTrainingProviderEditModelValidator.cs]: 👍

**Pull request merged**